components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java

```diff
@@ -319,6 +318,16 @@ public Response callback(@Context Request request, @PathParam("appName") String
                     log.debug("Set cookies for " + appName + " application.");
                 }
                 // Redirect to the store/apis page (redirect URL)
+                String uiServiceUrl;
+                //The first host in the list "allowedHosts" is the host of UI-Service
+                String uiServiceHost = APIMConfigurationService.getInstance().getApimConfigurations()
+                        .getEnvironmentConfigurations().getAllowedHosts().get(0);
+                if (uiServiceHost.isEmpty()) {
```

Comment:
Use stringutils.isEmpty here. If null value returned there will be null pointer exception throws

Reply:
Done with https://github.com/wso2/carbon-apimgt/pull/4896/commits/a887a18fff7578d20969d7146e46329a165ae57e